api/services/real.py

```diff
@@ -5,64 +5,98 @@
 from api.services.base import BaseNominativeQuery, BaseGISNQuery
 from httpx import HTTPStatusError, RequestError
 
+
 class RealNominativeQuery(BaseNominativeQuery):
 
     def fetch_data(self, street: str, house_number: int):
-      query_string = " ".join([street, str(house_number), "תל", "אביב"])
-      url = "https://nominatim.openstreetmap.org/search?"
-      params = {
-        "q": query_string,
-        "format": "json"
-      }
-
-      headers = {
-        "User-Agent": os.getenv('USER_AGENT'),
-        "Referer": os.getenv('REFERRER')
-      }
-
-      try:
-          response = httpx.get(url, params=params, headers=headers, timeout=5)
-          response.raise_for_status()
-          data = response.json()
-      except HTTPStatusError as e:
-          return JsonResponse({"error": f"Nominatim API error: {e.response.status_code} {e.response.reason_phrase}"}, status=e.response.status_code)
-      except RequestError:
-          return JsonResponse({"error": "Nominatim request failed"}, status=500)
-      except (ValueError, TypeError):
-          return JsonResponse({"error": "Invalid JSON response from Nominatim"}, status=500)
-
-      if not data:
-          return JsonResponse({"error": "could not locate address"}, status=500)
-
-      # collect all places
-      places = {
-          i: (place.get('lon'), place.get('lat'))
-          for i, place in enumerate(data)
-          if place.get('lat') and place.get('lon')
-      }
-
-      if not places:
-          return JsonResponse({"error": "No valid lat/lon found in Nominatim results"}, status=500)
-
-      return next(iter(places.values()))  # Get first available coordinates
+        query_string = " ".join([street, str(house_number), "תל", "אביב"])
+        url = "https://nominatim.openstreetmap.org/search"
+        params = {
+            "q": query_string,
+            "format": "json",
+        }
+
+        # Nominatim requires a User-Agent header (usage policy)
+        # Set defaults if not provided
+        user_agent = os.getenv('USER_AGENT', 'TamaOd/1.0 (https://github.com/your-org/tamaod)')
+        referer = os.getenv('REFERRER', 'https://tamaod.local')
+
+        # Build headers dict, only including non-None values
+        headers = {}
+        if user_agent:
```

Comment:
We have defaults for `user_agent` and `referer`, so these `if`s are not really necessary. We can still end up with the user setting e.g. `USER_AGENT` to an empty string, but this should be handled above:
```python
user_agent = os.getenv('USER_AGENT')
if not user_agent:
    user_agent = that_default_value
```